### DISCUSSION TITLE
Price suggestions are not working in for cycle

### DISCUSSION THREAD

**Discussion by yanshrimp** [joalla/discogs_client#97] [Support] [answered]
Hi!
I've got a problem with getting price suggestions.
It works well when you make just single request, but i get error when i do it in cycle.
```
results = d.search('my request using name')

for result in results:
    if 'master' in str(result).lower():  #cutting out master release
        continue
    release_id = str(result).split(' ')[1]  #getting id
    rel = d.release(release_id)
    price = rel.price_suggestions.very_good_plus.value
    print(price)
```

_____
53.950617283950606  # first iteration works well

Traceback (most recent call last):
  File "C:\Users\yan\Desktop\PYTHON_PROJECTS\PYTHON_PRACTICE_2022\REQUESTS_PRACTICE\discogs_price.py", line 24, in <module>
    price = rel.price_suggestions.very_good_plus.value
  File "C:\Users\yan\Desktop\PYTHON_PROJECTS\PYTHON_PRACTICE_2022\venv\lib\site-packages\discogs_client\models.py", line 26, in __get__
    value = instance.fetch(self.name)
  File "C:\Users\yan\Desktop\PYTHON_PROJECTS\PYTHON_PRACTICE_2022\venv\lib\site-packages\discogs_client\models.py", line 257, in fetch
    return self.data.get(key, default)
AttributeError: 'NoneType' object has no attribute 'get'

__________________

Also i'm curious is there any way to get data of sold items on particular release (lower and higher price)?
And prices data from "For sale" page for particular release.

I will be very grateful if someone can help me with this!!

**Comment by yanshrimp** [comment, marked as answer]
Got it!! Problem is because some of the releases don't have selling history and can't return any suggestions!

But questions about data of sold items and "for sale" prices are still actual.

**Reply by AnssiAhola** [reply]
Unfortunately the Discogs API lacks any kind of "general" marketplace features, ones that are useful anyway. 😅 
For example there's no way to search "For Sale" listings for a given release.

You can get users inventory (items for sale) and find all listings for a given release/release id
```python
inventory = client.user("username").inventory
for listing in inventory:
    if listing.release.id == 23227544:
        print(listing.price)
```
Although you have to go through them all to get the price data for specific releases, if that's what you wanted.

Also you can get the currently lowest price and amount for sale in marketplace for a given release
```python
release = client.release(23227544)
print(release.marketplace_stats.lowest_price)
print(release.marketplace_stats.num_for_sale)
```
I don't think there's a way to get the highest price for release though.

Hope that helped in some way.

Cheers,
-Anssi

**Reply by yanshrimp** [reply]
Thanks Anssi!
All the best!
